### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by Bouni**
I started adding more unit tests to get more coverage.

### CODE REVIEW

**Comment by Bouni** [comment]
I tried really hard to get rid of this [1%](https://github.com/Bouni/python-luxtronik/blob/0b2d91896827d61931fd12ad9d1e6f58bfdb8900/luxtronik/datatypes.py#L82) but had no luck so far.

There is something I don't understand about child classes and instanceof which always return the base class ...

@kbabioch Can you please take a look at this?

**Comment by gerw** [comment]
Can't you just compare a `Base("bar")` with something completely else, e.g., the integer `1`?

**Comment by Bouni** [comment]
I just don't understand what was the initial motivation for the check if the class is not equal to Base here.

In my opinion we could remove this check altogether.

A child class still has all three attributes that are compared below that if I understand it right.

@kbabioch can you give us some clarification of what your intention was with that check?

**Comment by gerw** [comment]
I think that the rational for the test `isinstance(other, Base)` is that if `other` is not an instance of `Base`, we cannot access `other.value`, `other.datatype_class` and `other.datatype_unit` in the lines below and this would raise an error.

The same comment would apply to `__lt__`.

However, `python3` itself raises a `TypeError` on `"barfoo" < 1`, but returns `False` on `"barfoo" == 1`. I think that we should follow the same principle?

**Comment by Bouni** [comment]
Ok, so I probably misundertood this. I thougt the check should return `False` if `other` is a subclass of Base, like `Frequency` for example.
So the `TypeError` comes from python but we need to take care of the return False for implemented methods like `__eq__`, `__lt__` and so on?

**Comment by gerw** [comment]
> So the `TypeError` comes from python

Which `TypeError` do you refer to?

> but we need to take care of the return False for implemented methods like `__eq__`, `__lt__` and so on?

I think that I am not sure what you mean.

From my side:

* The `__eq__` method is fine, we just need a test for the `return False`, e.g., `Base("barfoo") == 1`.
* The `__lt__` should be improved and should raise a `TypeError` if the two compared objects are not comparable, e.g., if `not isinstance(other, Base)` or if `self.datatype_class != other.datatype_class` or if `self.datatype_unit != other.datatype_unit`.

**Comment by Bouni** [comment]
I think I didn't understand what you iniztially wrote the right way but no I think I got it 😅 

So for `__eq__` we return false but for other methods like `__lt__`, `__gt__`  we raise an Exception instead.

But I think that should not be the scope of this PR but a seperate issue instead. I'll open one for that.

File: tests/test_datatypes.py [hunk 1]
```diff
@@ -94,25 +93,82 @@ def test_to_heatpump(self):
...
+
+        a = Base("base")
+        a.value = 1
+        assert a._raw == 1
+
+    def test_raw_property(self):
+        """Test case for raw property"""
+
+        a = Base("base")
+        a._raw = 1
```
[the remark this answers is not in the document]
Comment (Bouni) [reply tests/test_datatypes.py:122]: You mean different numbers for each test case or just this one e.g. random()?

[the remark this answers is not in the document]
Comment (gerw) [reply tests/test_datatypes.py:122]: I think fixed random numbers (in the spirit of https://xkcd.com/221/) should be ok :grin:

[the remark this answers is not in the document]
Comment (Bouni) [reply tests/test_datatypes.py:122]: 😂

**Comment by Bouni** [comment]
I think its better to create a new PR for more unit tests to make them easy to review.

**Review by gerw — APPROVED** [review]
LGTM. Should we squash the commits before merging the PR?

**Comment by Bouni** [comment]
@gerw Yes if you consider squashing helpful here, go for it (never used it before 😅 )

**Pull request merged**